rllib/algorithms/dreamerv3/tests/test_dreamerv3.py

```diff
@@ -0,0 +1,200 @@
+    @classmethod
+    def setUpClass(cls):
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        ray.shutdown()
+
+    def test_dreamerv3_compilation(self):
+        """Test whether DreamerV3 can be built with all frameworks."""
+
+        # Build a DreamerV3Config object.
+        config = (
+            dreamerv3.DreamerV3Config()
+            .framework(eager_tracing=True)
+            .training(
+                # Keep things simple. Especially the long dream rollouts seem
+                # to take an enormous amount of time (initially).
+                batch_size_B=2 * 2,  # shared w/ model AND learner AND env runner
+                batch_length_T=16,
+                horizon_H=5,
+
+                # TODO (sven): Fix having to provide this.
+                #  Should be compiled automatically as `RLModuleConfig` by
+                #  AlgorithmConfig (see comment below)?
+                model={
+                    "batch_length_T": 16,
+                    "horizon_H": 5,
+                    "model_size": "nano",  # Use a tiny model for testing.
+                    "gamma": 0.997,
+                    "symlog_obs": True,
+                },
+                _enable_learner_api=True,
+            )
+            .resources(
+                num_learner_workers=2,  # Try with 2 Learners.
+                num_cpus_per_learner_worker=1,
+                num_gpus_per_learner_worker=0,
+                num_gpus=0,
+            )
+            .debugging(log_level="INFO")
+            .rl_module(_enable_rl_module_api=True)
+        )
+
+        # TODO (sven): Add a `get_model_config` utility to AlgorithmConfig
+        #  that - for now - merges the user provided model_dict (which only
+        #  contains settings that only affect the model, e.g. model_size)
+        #  with the AlgorithmConfig-wide settings that are relevant for the model
+        #  (e.g. `batch_size_B`).
+        # config.get_model_config()
+
+        num_iterations = 2
+
+        for _ in framework_iterator(config, frameworks="tf2"):
+            for env in ["FrozenLake-v1", "CartPole-v1", "ALE/MsPacman-v5"]:
+                print("Env={}".format(env))
+                config.environment(env)
+                algo = config.build()
+
+                for i in range(num_iterations):
+                    results = algo.train()
+                    print(results)
+
+                algo.stop()
+
+    def test_dreamerv3_dreamer_model_sizes(self):
+        """Tests, whether the different model sizes match the ones reported in [1]."""
+
+        # For Atari, these are the exact numbers from the repo ([3]).
+        # However, for CartPole + size "S" and "M", the author's original code will not
+        # match on the world model count. This is due to the fact that the author uses
+        # encoder/decoder nets with 5x1024 nodes (which corresponds to XL) regardless of
+        # the `model_size` settings (iff >="S").
```

Comment:
Can we still check L and XL with respect to today's parameter's count to avoid regression in the future?

Reply:
ok, I can check, how much this would blow up the test, but yeah, that would make it more future proof, agreed.

Reply:
done